File: 0392-is-subsequence/0392-is-subsequence.kt
class Solution {
fun isSubsequence(s: String, t: String): Boolean {
    if (s.isBlank()) return true

    var currentChar = 0
    var subseq = ""
    for (c in t) {
        if (s[currentChar] == c) subseq += s[currentChar].also { currentChar++ }
        if (subseq == s) return true
    }

    return false
}
}

File: 0392-is-subsequence/0392-is-subsequence.go
func isSubsequence(s string, t string) bool {
	if len(s) == 0 {
		return true
	}

	// pointer u/ char yg akan di cek
	var currentChar uint8 = 0
	var subSequence string = ""

	for _, char := range t {
		if s[currentChar] == uint8(char) {
			subSequence += string(s[currentChar])
			currentChar++
		}

		if subSequence == s {
			return true
		}
	}

	return false
}